pint/models/frequency_dependent.py

```diff
@@ -0,0 +1,61 @@
+"""This module implements a frequency evolution of pulsar profiles model"""
+from warnings import warn
+import parameter as p
+from .timing_model import TimingModel, Cache
+import astropy.units as u
+import numpy as np
+import pint.utils as ut
+import astropy.time as time
+
+class FD(TimingModel):
+    """This class provides a timing model for frequency evolution of pulsar
+    profiles model.
+    """
+    def __init__(self):
+        super(FD, self).__init__()
+        self.add_param(p.prefixParameter(name='FD1', units="second", value=0.0,
+                       descriptionTplt=lambda x: ("%d term of frequency"
+                                                  " dependent  coefficients" % x),
+                       unitTplt=lambda x: 'second',
+                       type_match='float'))
+
+        self.delay_funcs['L1'] += [self.FD_delay]
```

Comment:
Here, it would be good to put in a comment about why these delays were being assigned to 'L1'.
As I understand, 'L1' is for delays local to the Solar System that should be applied to get (what we usually call) barycenter-corrected TOAs.  I'm actually not sure whether FD terms should be L1 or L2.

This also made me realize that L1 and L2 are not very descriptive names.  We should discuss a better naming convention.  Maybe 'SolarSystem' and 'PulsarSystem' or something like that?

Also, should the FD terms really be a phase lag, rather than a delay term?  I guess they are implemented as delays in TEMPO.  What do you think @demorest ?